pkg/util/docker/global.go

```diff
@@ -211,20 +211,41 @@ func ResolveImageName(image string) (string, error) {
 	if globalDockerUtil != nil {
 		return globalDockerUtil.extractImageName(image), nil
 	}
-	return "", errors.New("dockerutil not initialised")
+	return image, errors.New("dockerutil not initialised")
 }
 
-// SplitImageName splits a valid image name (from ResolveImageName)
-// into the name and tag parts.
-func SplitImageName(image string) (string, string, error) {
+// SplitImageName splits a valid image name (from ResolveImageName) and returns:
+//    - the "long image name" with registry and prefix, without tag
+//    - the "short image name", without registry, prefix nor tag
+//    - the image tag if present
+//    - an error if parsing failed
+func SplitImageName(image string) (string, string, string, error) {
+	// See TestSplitImageName for supported formats (number 6 will surprise you!)
 	if image == "" {
-		return "", "", errors.New("empty image name")
+		return "", "", "", errors.New("empty image name")
 	}
-	parts := strings.SplitN(image, ":", 2)
-	if len(parts) < 2 {
-		return image, "", errors.New("could not find tag")
+	long := image
+	if pos := strings.LastIndex(long, "@sha"); pos > 0 {
```

Comment:
If SplitImageName is called after ResolveImageName (from computeDockerIDs). Images should not carry the sha as they are resolved to their friendly name https://github.com/DataDog/datadog-agent/blob/aa3af46e52208cf83da6a1f9d4fa7ac0a0365dc5/pkg/util/docker/docker_util.go#L363 isn't it ?
So the long var should not be used ?

Reply:
This addresses this case (from swarm): https://github.com/DataDog/datadog-agent/pull/776/files/e75d803d812dd35448db46285051fc32913c86c4#diff-f81a6ea4ba07c049b39e7f74dc224a2dR41

We might be able to handle it in ResolveImageName, but we currently don't , and better twice than never when it comes to these edge cases